### DISCUSSION TITLE
Is there anyone who understands what are the advantages of this type of library?

### DISCUSSION THREAD

**Discussion by esbylan** [vue-styled-components/core#26] [General]
以前没用过这类css in js的库，看了下文档觉得写法很新颖很酷，调用方式看起来也很清晰直观，还有就是维护某些动态样式可能会比较舒服，除此之外 有没有重度使用者讲讲这个优势或者典型应用场景是什么（比如相比于直接写css或者sass、less，或者动态绑定行内样式，是否存在某个场景，用这类写法能让代码更优雅或者更易维护）(´･_･`)

[a comment by an automated account is not shown]

**Comment by akinoccc** [comment]
emmm...我感觉使用在ui库开发可能更合适，css in js适合组合式调用，之前有去开发过一个组件库，用的sass，变量维护起来特别灾难，各种变量嵌套/组合使用起来都特别麻烦

**Comment by esbylan** [comment]
> emmm...我感觉使用在ui库开发可能更合适，css in js适合组合式调用，之前有去开发过一个组件库，用的sass，变量维护起来特别灾难，各种变量嵌套/组合使用起来都特别麻烦

有道理，仔细想想如果无脑抛弃写样式表，把整个页面的样式全部写在js，那js部分就特别臃肿了，然后大部分需求用简单的动态样式就能搞定，用这类css in js有点大炮打蚊子的意味，确实比较适合组件库开发

**Comment by akinoccc** [comment]
> > emmm...我感觉使用在ui库开发可能更合适，css in js适合组合式调用，之前有去开发过一个组件库，用的sass，变量维护起来特别灾难，各种变量嵌套/组合使用起来都特别麻烦
> 
> 有道理，仔细想想如果无脑抛弃写样式表，把整个页面的样式全部写在js，那js部分就特别臃肿了，然后大部分需求用简单的动态样式就能搞定，用这类css in js有点大炮打蚊子的意味，确实比较适合组件库开发

对于业务而言，用来封装通用业务组件还是挺爽的，不过这个就见仁见智了哈哈

**Comment by esbylan** [comment]
> 对于业务而言，用来封装通用业务组件还是挺爽的，不过这个就见仁见智了哈哈

懂了，这让我联想起某文档的一段话😄
![image](https://github.com/v-vibe/vue-styled-components/assets/31299495/095bdb28-3af4-4626-9579-005466db2cbc)

**Comment by linlifeng1995** [comment]
Great topic, I've been confused about this too. I feel enlightened now, thank you.

**Comment by akinoccc** [comment]
> Great topic, I've been confused about this too. I feel enlightened now, thank you.

It's happy to be able to help you！😀
